### ISSUE TITLE
Boot splash doesn't work on mobile

### ISSUE THREAD

**Issue by oeleo1** [godotengine/godot#38897]
**Godot version:**
3.2.2.beta2

**OS/device including version:**
Android, iOS

**Issue description:**
Under circumstances to be determined, the boot splash screen is consistently not showing on startup for mobile devices, which is unacceptable.  There are open issues mentioning the problem, but I am opening a new issue to give this more weight and primarily to provide a minimal project which reproduces the problem reliably for those devices where this is true.

The project is a 2 node main scene where one load node is defined in terms of 32k pass statements keeping the parser/bytecode compiler busy for some time until it's done, then the second label node is shown which attests for the end of the startup sequence.

It's not a GLES2 vs GLES3 thing, because I have a project where the boot splash is showing consistently with GLES3 while not showing consistently with GLES2 on both Android and iOS, but the attached minimal project triggers the issue reliably on both GLES2/3 for relevant devices (presumably weaker ones).

Someone with good understanding of the boot sequence shall have a look. An uneducated guess is that in case it's a multi-threading issue, the visual server doesn't have a chance to run while the parser is fully loaded. Which explains the sporadic behavior of the problem for some devices and not for others. Yet, I could be completely wrong.

**Steps to reproduce:**
Try the minimal project on a mobile device.

**Minimal reproduction project:**
[Boot_Splash.zip](https://github.com/godotengine/godot/files/4658418/Boot_Splash.zip)

**Comment by Calinou** [comment]
Duplicate of #21824.

> There are open issues mentioning the problem, but I am opening a new issue to give this more weight and primarily to provide a minimal project which reproduces the problem reliably for those devices where this is true.

Please upload a minimal reproduction project in a comment on the original issue instead. This makes it easier to track which issues have been solved and which ones haven't :slightly_smiling_face:

**Comment by oeleo1** [comment]
Okay, done. But I'll quote myself here :)

ROOOOOAAARRRRR !!! :-)

This is just in case Godot needs a wake-up call for a critical issue for any game maker who respects herself on an issue opened 2 years ago with no assignee, no milestone, and with a proven record of being forgotten.

Jesus, Godot can't even show its splash screen reliably...

**Comment by Calinou** [comment]
@oeleo1 I have to remind you that we have a [Code of Conduct](https://godotengine.org/code-of-conduct). Please stay constructive.

**Comment by oeleo1** [comment]
Indeed. So when I am opening a new issue on purpose and explain why I am doing so by contributing the missing element  to resolve the issue, and see that the issue is being closed within the hour by redirecting me to the one I wanted to avoid in the first place due its track record, the problem is not on my end and you deserve a reaction.

I in turn have to draw your attention to the fact that in this situation your smiley and explanation were as displaced as the reaction you got and actually caused it. Since you are mentioning the Code of Conduct which I am fully aware of and aligned with, feel free to submit the case to the Committee in charge of its application.

That being said, I am interested in making progress here instead of arguing, Submitting a new issue with a reproduction project is constructive. My observation of the process is that the team/you crumble under load to deal with tons of bugs reports about micro-issues compared to this one and naturally things fade away. So there is this weakness (but also strength) in the process of natural selection, where bugs go into oblivion unless their have higher priority and we have enough evidence and understanding to fix them.

This is such a case. What does it take in your opinion or what should I have done in order to bring this issue back on the table?  You know better than me that the signal/noise ratio is such that a comment in the old issue wouldn't have made a difference so that's not an answer.

Case closed. Thanks.

**Comment by groud** [comment]
> the problem is not on my end and you deserve a reaction

There nothing like "you" here, this is a fully open source project built by the community. The engine is as much yours as it is mine. 

Today, the one having an issue is you, if you think it's very important for your project, fix the issue yourself, and if you can't do it, pay someone to do it. 

And if you still cannot do that, then just don't ask people around to work for free for you. It's not how open source work.

This bug tracker is only here to help contributors understand problems and select issues they might be interested in, it's not a place for users to complain.

**Comment by oeleo1** [comment]
@groud Calinou was too quick at shutting this down for no real reason because he is as always a terrific front-end 1st pass bug filter. I reacted because after giving it a thought I opened a new issue which is probably different in behavior, scope, cause, Godot version, etc. than the one he referred me to. I have the right to disagree with his decision which resulted in a funny exchange. As a result, you felt compelled lecturing me off-topic. Well, I can't stop you from doing that but usually open source people don't ramble like this. If you are trying to say that you are up to the task and would like to be paid for the fix, that's a different thing and we are on-topic again. I'd be happy to pay you for this particular fix, assuming "you" exists around here of course. Your price will be mine, unless, like you say, we fix it by ourselves in the meantime. Godot has many (known) dark corners but we always found our way around those with crispy workarounds (object.queue_free(), is_instance_valid(), hiccups due to shader compilation, etc., etc.) but this one is a system issue with no known workaround and therefore is considered a show stopper. It's even worse as we don't know exactly what is causing it. It may be anything from threads to rendering or drivers...
